hack/build-farm-canary.sh

```
@@ -0,0 +1,47 @@
+        cases+="  <testcase name=\"${name}\" classname=\"build-farm-canary\"><failure/></testcase>\n"
+        echo "FAIL: ${name}" >&2
+        rc=1
+    fi
+}
+
+http_ok() {
+    local url=$1 timeout=${2:-30}
+    local s
+    s=$(curl -ksS --max-time "${timeout}" -o /dev/null -w "%{http_code}" "${url}" 2>/dev/null)
+    [[ "$s" == "200" || "$s" == "401" ]]
+}
```

Shellcheck (0.11.0)

[info] This function is never invoked. Check usage (or ignored if invoked indirectly).

(SC2329)